common/meta/api/src/schema_api_impl.rs

```diff
@@ -192,6 +194,81 @@ impl<KV: KVApi> SchemaApi for KV {
         }
     }
 
+    async fn rename_database(
+        &self,
+        req: RenameDatabaseReq,
+    ) -> Result<RenameDatabaseReply, MetaError> {
+        let tenant_dbname = &req.name_ident;
+        let tenant_newdbname = DatabaseNameIdent {
+            tenant: tenant_dbname.tenant.clone(),
+            db_name: req.new_db_name.clone(),
+        };
+
+        loop {
+            // get old db, not exists return err
+            let res = get_db_or_err(
+                self,
+                tenant_dbname,
+                format!("rename_database: {}", &tenant_dbname),
+            )
+            .await;
+
+            let (_, old_db_id, _, old_db_meta) = match res {
+                Ok(x) => x,
+                Err(e) => {
+                    if let MetaError::AppError(AppError::UnknownDatabase(_)) = e {
+                        if req.if_exists {
+                            // TODO(like rename table): if database not exists but sql hint if_exists is true
+                            // the sql must be Ok but should not return database_id
+                            return Ok(RenameDatabaseReply { db_id: 0 });
```

Comment:
I'd suggest removing `db_id` in the `RenameDatabaseReply` because renaming a db does not change its id.

Reply:
Ok I will have a look.